inc/settings/class-settings-misc.php

```diff
@@ -47,17 +47,54 @@ public function get_settings() {
 					'checkboxgroup' => 'start',
 					'desc_tip'      => __( 'We guarantee no sensitive data is collected. ', 'ang' ) . '<a class="ang-link" href="https://docs.analogwp.com/article/547-what-data-is-tracked-by-the-plugin" target="_blank">' . __( 'More Info', 'ang' ) . '</a>',
 				),
+				array(
+					'type' => 'sectionend',
+					'id'   => 'ang_misc',
+				),
+
+				array(
+					'type' => 'title',
+					'id'   => 'sk_uninstall',
+				),
 				array(
 					'title'         => __( 'Remove Data on Uninstall', 'ang' ),
-					'desc'          => __( 'Check this box to remove all data stored by Style Kit for Elementor plugin, including license info, user settings, import history etc. Any imported or manually saved Style Kits are not removed.', 'ang' ),
+					'desc'          => __( 'Toggle this to enable plugin data removal upon uninstall.', 'ang' ),
+					'class'         => 'sk-uninstall',
 					'id'            => 'remove_on_uninstall',
 					'default'       => false,
 					'type'          => 'checkbox',
 					'checkboxgroup' => 'start',
+					'is_option'     => false,
 				),
 				array(
 					'type' => 'sectionend',
-					'id'   => 'ang_misc',
+					'id'   => 'sk_uninstall',
+				),
+
+				array(
+					'type' => 'title',
+					'id'   => 'sk_uninstall_options',
+				),
+				array(
+					'class'   => 'ang-uninstall-options',
+					'desc'    => __( 'You can multi-select from the options below and upon plugin uninstall these will be respected.', 'ang' ),
+					'id'      => 'uninstall_options',
+					'default' => array(
+						'remove_all_media' => true,
+						'plugin_settings'  => true,
+					),
+					'type'    => 'multi-checkbox',
+					'options' => array(
+						'remove_all_media' => __( 'Remove all media files imported along Templates and Blocks', 'ang' ),
```

Comment:
Remove 'Remove all media files imported along with Templates and Blocks' option, currently, there is no way to identify imported images I'm afraid. Any imported images would appear as imported by Elementor because we use their core functions to import images.